statsd/client_test.go

```diff
@@ -150,16 +150,16 @@ func TestBufferedClient(t *testing.T) {
 
 	dur, _ := time.ParseDuration("123us")
 
-	client.Incr("ic", nil, 1)
-	client.Decr("dc", nil, 1)
-	client.Count("cc", 1, nil, 1)
-	client.Gauge("gg", 10, nil, 1)
-	client.Histogram("hh", 1, nil, 1)
-	client.Distribution("dd", 1, nil, 1)
-	client.Timing("tt", dur, nil, 1)
-	client.Set("ss", "ss", nil, 1)
-
-	client.Set("ss", "xx", nil, 1)
+	client.Incr("ab", nil, 1)
```

Comment:
although you do it on the benchmark test, would we want to have a different test with more random metric naming to ensure we test the actual sharding of the metrics to the different workers?

Reply:
Not sure it's worth it:
- Testing with arbitrary value is not very valuable I think
- Testing with random value will cause the test to be flaky

Since it's not part of the public API I don't think unit test should cover this part. The  benchmark are better suited for this I think.

Reply:
You're absolutely right, randomizing the metric name would make the test unreliable. Sounds good to just do this on the benchmark test.